pkg/runner/cluster_k8s.go

```diff
@@ -318,46 +391,66 @@ func (c *ClusterK8sRunner) Run(ctx context.Context, input *api.RunInput, ow *rpc
 		}
 	}
 
-	err = eg.Wait()
-	if err != nil {
-		return nil, err
-	}
+	// we want to fetch logs even in an event of error
+	defer func() {
+		if input.TotalInstances <= 200 {
+			var gg errgroup.Group
 
-	if input.TotalInstances <= 500 {
-		var gg errgroup.Group
+			for _, g := range input.Groups {
+				for i := 0; i < g.Instances; i++ {
+					i := i
+					g := g
+					sem <- struct{}{}
 
-		for _, g := range input.Groups {
-			for i := 0; i < g.Instances; i++ {
-				i := i
-				g := g
-				sem <- struct{}{}
+					gg.Go(func() error {
+						defer func() { <-sem }()
 
-				gg.Go(func() error {
-					defer func() { <-sem }()
+						podName := fmt.Sprintf("%s-%s-%s-%d", jobName, input.RunID, g.ID, i)
 
-					podName := fmt.Sprintf("%s-%s-%s-%d", jobName, input.RunID, g.ID, i)
+						ow.Debugw("fetching logs", "pod", podName)
+						logs, err := c.getPodLogs(ow, podName)
+						if err != nil {
+							return err
+						}
+						ow.Debugw("got logs", "pod", podName, "len", len(logs))
 
-					logs, err := c.getPodLogs(ow, podName)
-					if err != nil {
+						_, err = ow.WriteProgress([]byte(logs))
```

Comment:
For now we forward `stdout` logs from test plan instances to the `logs` output for given task. If test plans are well-implemented, in the future we might not need to do that.

Reply:
With proper run events in the sync service now, we should stop doing this. The fact that the runner behaves different with different pod counts is confusing. We should also get rid of the tailing.

Reply:
I both agree and disagree with you:
a) agree because this is really hacky and confusing and not consistent across runners.
b) disagree about removing it now, because we will lose stdout and it won't be available to users, and this will make it much harder to debug test plans, that are incorrectly using stdout for various debug messages and errors.

I'd rather have a better UX at the moment, and slightly harder and complicated codebase, than well architected runner and bad UX.